baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/GuavaPreconditionsEnforceLogSafeUse.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.regex.Pattern;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "GuavaPreconditionsEnforceLogSafeUse",
+        category = BugPattern.Category.ONE_OFF,
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "Guava Preconditions.checkX(Object) and Preconditions.checkX(Object, String) methods with a constant "
+                + "message and no parameters should use logsafe Preconditions.checkX(Object, String)")
+public final class GuavaPreconditionsEnforceLogSafeUse extends EnforceLogSafeUseFormat {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Matcher<ExpressionTree> PRECONDITIONS_METHOD =
+            Matchers.anyOf(
+                    MethodMatchers.staticMethod()
+                            .onClass("com.google.common.base.Preconditions")
+                            .withNameMatching(Pattern.compile("checkArgument|checkState|checkNotNull")));
```

Comment:
Probably a separate PR, but would be good to check Guava’s [`Verify.verify` & `Verify.verifyNotNull`](https://google.github.io/guava/releases/23.1-jre/api/docs/com/google/common/base/Verify.html) as well as [commons-lang `Validate.*`](https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/Validate.html)?

Reply:
That would be difficult -- guava Preconditions throws standard IllegalState/IllegalArg/NullPointer exceptions, but these methods throw library-specific exception types, so a migration _could_ break existing catches.

I think we should suggest against using them in favor of safe-logging preconditions, but I'd prefer to do so in a different check.

Reply:
👍 Makes sense for Guava's `Verify` and I expect much lower usage of that.

The commons-lang `Validate` is much more heavily used in some places, and those should be straightforward replacements by equivalent `Safe` preconditions

Reply:
I had though lang3 validate used custom exception types, thanks for the correction! In that case it makes sense to add common methods to this check. Let's add it in a separate PR.